monitor/internal/pod/delete_controller.go

```diff
@@ -114,7 +117,13 @@ func deleteControllerProcessItem(backgroundCtx context.Context, c client.Client,
 
 	// the edge case: a pod with the same namespaced name came up and we have missed a delete event
 	// this means that this pod belongs to a different PU and must live, therefore we try to delete the old one
-	if nativeID != string(pod.GetUID()) {
+	//if nativeID != string(pod.GetUID()) {
+	currentPuID, err := sandboxExtractor(ctx, pod)
```

Comment:
I wonder ... what does this mean in the case that the pod and/or sandbox are already gone?

Reply:
I think we'll probably have to make that check now a triple: namespacedName && puID && sandboxID 

However, I might be wrong :) ... Just food for thought

Reply:
Awesome point, let me think over it and make changes.